app/main.py

```diff
@@ -1,52 +1,116 @@
         self.content = content
 
-    def display(self, display_type: str) -> None:
-        if display_type == "console":
-            print(self.content)
-        elif display_type == "reverse":
-            print(self.content[::-1])
-        else:
-            raise ValueError(f"Unknown display type: {display_type}")
-
-    def print_book(self, print_type: str) -> None:
-        if print_type == "console":
-            print(f"Printing the book: {self.title}...")
-            print(self.content)
-        elif print_type == "reverse":
-            print(f"Printing the book in reverse: {self.title}...")
-            print(self.content[::-1])
-        else:
-            raise ValueError(f"Unknown print type: {print_type}")
-
-    def serialize(self, serialize_type: str) -> str:
-        if serialize_type == "json":
-            return json.dumps({"title": self.title, "content": self.content})
-        elif serialize_type == "xml":
-            root = ET.Element("book")
-            title = ET.SubElement(root, "title")
-            title.text = self.title
-            content = ET.SubElement(root, "content")
-            content.text = self.content
-            return ET.tostring(root, encoding="unicode")
-        else:
-            raise ValueError(f"Unknown serialize type: {serialize_type}")
+    def display(self, strategy: DisplayStrategy):
+        strategy.display(self)
+
+    def print_book(self, strategy: PrintStrategy):
+        strategy.print_book(self)
+
+    def serialize(self, strategy: SerializeStrategy):
+        return strategy.serialize(self)
+
+
+def get_display_strategy(display_type):
+    if display_type == "console":
+        return ConsoleDisplay()
+    elif display_type == "reverse":
+        return ReverseDisplay()
+    else:
+        raise ValueError(f"Unknown display type: {display_type}")
+
+
+def get_print_strategy(print_type):
+    if print_type == "console":
+        return ConsolePrint()
+    elif print_type == "reverse":
+        return ReversePrint()
+    else:
+        raise ValueError(f"Unknown print type: {print_type}")
+
+
+def get_serialize_strategy(serialize_type):
+    if serialize_type == "json":
+        return JsonSerializer()
+    elif serialize_type == "xml":
+        return XmlSerializer()
+    else:
+        raise ValueError(f"Unknown serialize type: {serialize_type}")
 
 
 def main(book: Book, commands: list[tuple[str, str]]) -> None | str:
     for cmd, method_type in commands:
         if cmd == "display":
-            book.display(method_type)
+            strategy = get_display_strategy(method_type)
+            book.display(strategy)
         elif cmd == "print":
-            book.print_book(method_type)
+            strategy = get_print_strategy(method_type)
+            book.print_book(strategy)
         elif cmd == "serialize":
-            return book.serialize(method_type)
+            strategy = get_serialize_strategy(method_type)
+            return book.serialize(strategy)
```

Comment:
The `main` function returns the result of the first 'serialize' command it encounters. If there are multiple commands and you want to process all of them, consider storing the results in a list and returning the list at the end. This way, you can handle multiple 'serialize' commands and return their results collectively.